### DISCUSSION TITLE
Fast Refresh on StaticProps change

### DISCUSSION THREAD

**Discussion by ervinoro** [vercel/next.js#15521] [Help]
Let's say my getStaticProps reads data from a local file (e.g. json) and I start the development server. I noticed that when I then change the json file, 

1. The development server notices and does something (wait  - compiling... event - compiled successfully)
2. Fast Refresh isn't triggered on the browser.

I understand that monitoring for prop changes can't be done in general, but in this particular (and I guess not too uncommon) scenario it could be done. So I wanted to ask: is this just not implemented, or have I misconfigured something and it is possible to get Fast Refresh working with data file changes as well?

**Comment by spaceemotion** [comment]
I actually am getting the fast refresh, but the static props are not being re-evaluated. The tab needs to be refreshed manually.

**Comment by aralroca** [comment]
Same problem. Any fix for this?

**Comment by kodai3** [comment]
In managing the locale json, I encountered this problem too.
Is there a workaround or fix?

Considering use cases like translation, wouldn't it be better to treat this as an issue rather than a discussion, should we change it? @ervinoro
